src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/AssessmentEntity.kt

```diff
@@ -185,8 +158,17 @@ interface AssessmentRepository : JpaRepository<AssessmentEntity, UUID> {
            join applications ap on a.application_id = ap.id
            left outer join temporary_accommodation_applications taa on ap.id = taa.id
      where taa.probation_region_id = ?1
-           and ap.crn = ?2
+           and (?2 is null or ap.crn = ?2)
            and a.reallocated_at is null
+           and (?3 is null OR 
```

Comment:
[I noted that we don't think this query will fall foul of the same problems as booking search.](https://mojdt.slack.com/archives/C03K0HB0LBE/p1702386439174639?thread_ts=1702383217.041969&cid=C03K0HB0LBE)

[It was interesting to learn that CAS1 have taken a different approach here with a new status column](https://github.com/ministryofjustice/hmpps-approved-premises-api/pull/1119). [They appear to be tackling the problem in the same way as our other idea](https://github.com/ministryofjustice/hmpps-approved-premises-api/pull/1119/files#diff-f5add0cc3f1580e3f29dbb37dda68174daacd19376c2d5b86e2c8ffd7ca11c77R3) but as we're going in second we're actually now in the position of introducing diverging behaviour.

Given that work, what do you think is best for the codebase as a whole?

I wonder if it's:

1. merge this as is for short term performance improvements
2. migrate to using a status column to make our handling of status more explicit and consistent

Reply:
We could make changes to create new status column to make our sql straight forward, but this will take time as we need to understand flow/journey of assessment and may take time to fix it.

I personally think, we can go with current work and then we can plan for adding the status later. I even thinking whether merge this code now or after christmas when we both wont be available if any issue.